controllers/authenticate.go

```diff
@@ -21,55 +21,44 @@ func NewAuthenticateController(getProfile func(string) (*auth0.Profile, error))
 	}
 }
 
-func (a *AuthenticateController) Routes() []*fireball.Route {
-	return []*fireball.Route{
-		{
-			Path: "/authenticate",
-			Handlers: fireball.Handlers{
-				"POST": a.authenticate,
-			},
-		},
-	}
-}
-
-func (a *AuthenticateController) authenticate(c *fireball.Context) (fireball.Response, error) {
-	defer c.Request.Body.Close()
-
-	var tr authentication.TokenReview
-	if err := json.NewDecoder(c.Request.Body).Decode(&tr); err != nil {
+func (a *AuthenticateController) Authenticate(r *http.Request) http.Handler {
+	defer r.Body.Close()
+	var tr auth.TokenReview
+	if err := json.NewDecoder(r.Body).Decode(&tr); err != nil {
 		log.Printf("[ERROR] Failed to decode TokenReview: %v", err)
-		return newTokenReviewResponse(authentication.TokenReviewStatus{Error: err.Error()})
+		return tokenReview(auth.TokenReviewStatus{Error: err.Error()})
 	}
 
 	profile, err := a.getProfile(tr.Spec.Token)
 	if err != nil {
-		return newTokenReviewResponse(authentication.TokenReviewStatus{Error: err.Error()})
+		log.Printf("[ERROR] Failed to get profile for token '%s': %v", tr.Spec.Token, err)
+		return tokenReview(auth.TokenReviewStatus{Error: err.Error()})
 	}
 
-	status := authentication.TokenReviewStatus{
+	trs := auth.TokenReviewStatus{
 		Authenticated: true,
-		User: authentication.UserInfo{
+		User: auth.UserInfo{
 			UID:      profile.Email,
 			Username: profile.Email,
 		},
 	}
 
-	return newTokenReviewResponse(status)
+	return tokenReview(trs)
 }
 
-func newTokenReviewResponse(status authentication.TokenReviewStatus) (fireball.Response, error) {
-	statusCode := http.StatusUnauthorized
-	if status.Authenticated {
-		statusCode = http.StatusOK
+func tokenReview(trs auth.TokenReviewStatus) http.Handler {
+	status := http.StatusUnauthorized
+	if trs.Authenticated {
+		status = http.StatusOK
 	}
 
-	resp := authentication.TokenReview{
-		TypeMeta: metav1.TypeMeta{
-			APIVersion: "authentication.k8s.io/v1beta1",
+	tr := auth.TokenReview{
+		TypeMeta: meta.TypeMeta{
+			APIVersion: "auth.k8s.io/v1beta1",
 			Kind:       "TokenReview",
 		},
-		Status: status,
+		Status: trs,
 	}
 
-	return fireball.NewJSONResponse(statusCode, resp)
+	return rye.JSON(status, tr)
```

Comment:
[rye nit] "JSON" is a bit ambiguous here. JSON what? By contrast, `Error` is both a noun and (maybe mistakenly) a verb, it's purpose is a bit more intuitive.

Reply:
I purposely chose than naming pattern (e.g. `JSON`, `Bytes`, `String`) because it's concise,  reads well, and accurately describes what is happening:
```go
return rye.JSON(...)  // reads as "return json", or "return a json response"
```

I think the ambiguity here would be alleviated  by looking at the the parameter names for [that function](https://godoc.org/github.com/zpatrick/rye#JSON) and by adding documentation to the `rye` package (which is coming - still WIP).